datafusion/expr/src/expr.rs

```diff
@@ -244,6 +244,8 @@ pub enum Expr {
     /// List of grouping set expressions. Only valid in the context of an aggregate
     /// GROUP BY expression list
     GroupingSet(GroupingSet),
+    /// A place holder for parameters in a prepared statement.
+    Placeholder(String),
```

Comment:
@NGA-TRAN  -- I was thinking about how to get the data type information from the SQL planner to here

I was wondering what if you looked up the placeholder during planning and then it would be available for the rest of the system

Something like 

```suggestion
    Placeholder {
      /// The identifier
      id: String,
      /// The type the parameter will be filled in with
      data_type: DataType,
     },
```

Reply:
I think this is a good idea. This way, we won't miss to get data type of any params

Reply:
I am trying to do this and see if I can eliminate the work in get data type that will simplify this PR a lot

Reply:
Done